drivers/spi/spi_ll_stm32.c

```diff
@@ -272,6 +272,60 @@ static int spi_dma_move_buffers(const struct device *dev, size_t len)
 /* Value to shift out when no application data needs transmitting. */
 #define SPI_STM32_TX_NOP 0x00
 
+static void spi_stm32_send_next_frame(SPI_TypeDef *spi,
+		struct spi_stm32_data *data)
+{
+	const uint8_t frame_size = SPI_WORD_SIZE_GET(data->ctx.config->operation);
+	uint32_t tx_frame = SPI_STM32_TX_NOP;
+
+	switch (frame_size) {
+	case 8:
+		if (spi_context_tx_buf_on(&data->ctx)) {
+			tx_frame = UNALIGNED_GET((uint8_t *)(data->ctx.tx_buf));
+		}
+		LL_SPI_TransmitData8(spi, tx_frame);
+		spi_context_update_tx(&data->ctx, 1, 1);
+		break;
+	case 16:
+		if (spi_context_tx_buf_on(&data->ctx)) {
+			tx_frame = UNALIGNED_GET((uint16_t *)(data->ctx.tx_buf));
+		}
+		LL_SPI_TransmitData16(spi, tx_frame);
+		spi_context_update_tx(&data->ctx, 2, 1);
+		break;
+	default:
```

Comment:
> you don't need that. Once you configured your controller, frame size will not change.

You mean the `switch` statement? I replaced the `if ... else ...` by `switch` because it leaves the code in a better position to support more frame sizes in the future (H7 has quite a lot of them). However, I don't mind reverting this back to `if ... else ...`.

Reply:
Frame size can be of below a byte or above, but less than 2 bytes etc... (I.E.: 4, 6, ... 10, 12...). But in the end you handle everything on byte-basis. So even if it's below a byte: you feed one byte (the controller will only take the relevant n bits representing the frame). Above a byte but below 2bytes: you feed 2 bytes. etc...

Stm32's HAL does not seem to provide anything but LL_SPI_TransmitData\<8/16/32\>. 

if you enable sizes above 16bits, then perhaps keep the switch. if/else if/else is less clean than a switch in such case.

Reply:
Agreed, I will revert to `if else`.

Reply:
Done